build/media-suite_helper.sh

```diff
@@ -1080,21 +1090,19 @@ do_removeOptions() {
 do_patch() {
     local binarypatch="--binary"
     case $1 in -p) binarypatch="" && shift ;; esac
-    local patch=${1%% *}       # Location or link to patch.
-    local am=$2                # Use git am to apply patch. Use with .patch files
-    local strip=${3:-1}        # Leading directories to strip. "patch -p${strip}"
+    local patch="${1%% *}"     # Location or link to patch.
     local patchName="${1##* }" # Basename of file. (test-diff-files.diff)
+    local am=false             # Use git am to apply patch. Use with .patch files
+    local strip=${3:-1}        # Leading directories to strip. "patch -p${strip}"
     [[ $patchName == "$patch" ]] && patchName="${patch##*/}"
+    [[ $2 == am ]] && am=true
 
-    if [[ -z $patchName ]]; then
-        # hack for URLs without filename
-        patchName="$(/usr/bin/curl -sI "$patch" | grep -Pio '(?<=filename=)(.+)')"
-        if [[ -z $patchName ]]; then
-            echo -e "${red}Failed to apply patch '$patch'${reset}"
-            echo -e "${red}Patch without filename, ignoring. Specify an explicit filename.${reset}"
-            return 1
-        fi
-    fi
+    # hack for URLs without filename
+    patchName=${patchName:-"$(/usr/bin/curl -sI "$patch" | grep -Eo 'filename=.*$' | sed 's/filename=//')"}
+    [[ -z $patchName ]] &&
+        printf '%b\n' "${red}Failed to apply patch '$patch'${reset}" \
```

Comment:
If the second line's also red, there's no need to reset, right?

Reply:
Remove extra reset and red